packages/datadog-plugin-tedious/src/index.js

```diff
@@ -0,0 +1,133 @@
+
+const procnameRegex = /^sp_[a-z]+$/
+
+function createWrapRequestClass (tracer) {
+  return function wrapRequestClass (Request) {
+    class RequestWithTrace extends Request {
+      constructor (sqlTextOrProcedure, callback) {
+        super(sqlTextOrProcedure, callback)
+        tracer.scope().bind(this)
+      }
+    }
+
+    return RequestWithTrace
+  }
+}
+
+function createWrapConnectionClass (tracer) {
+  return function wrapConnectionClass (Connection) {
+    class ConnectionWithTrace extends Connection {
+      constructor (config) {
+        super(config)
+        tracer.scope().bind(this)
+      }
+    }
+
+    return ConnectionWithTrace
+  }
+}
+
+function createWrapMakeRequest (tracer, config) {
+  return function wrapMakeRequest (makeRequest) {
+    return function makeRequestWithTrace (request) {
+      const connectionConfig = this.config
+      const scope = tracer.scope()
+      const childOf = scope.active()
+      const query = getQuery(request)
+
+      if (!query) {
+        return makeRequest.apply(this, arguments)
+      }
+
+      const span = tracer.startSpan(`tedious.request`, {
+        childOf,
+        tags: {
+          [Tags.SPAN_KIND]: Kinds.CLIENT,
+          'db.type': 'mssql',
+          'span.type': 'sql',
+          'component': 'tedious',
+          'service.name': config.service || `${tracer._service}-mssql`,
+          'resource.name': query
+        }
+      })
+
+      addConnectionTags(span, connectionConfig)
+      addDatabaseTags(span, connectionConfig)
+      addProcIdTags(span, request)
+
+      analyticsSampler.sample(span, config.analytics)
+      request.callback = tx.wrap(span, request.callback)
+
+      return scope.bind(makeRequest, span).apply(this, arguments)
+    }
+  }
+}
+
+function createWrapGetRowStream (tracer) {
+  return function wrapGetRowStream (getRowStream) {
+    return function getRowStreamWithTrace () {
+      const scope = tracer.scope()
+
+      const rowToPacketTransform = getRowStream.apply(this, arguments)
+      return scope.bind(rowToPacketTransform)
+    }
+  }
+}
+
+function getQuery (request) {
+  if (request.parameters) {
+    if (request.parameters.length === 0) {
+      return request.sqlTextOrProcedure
+    } else {
+      const statement = request.parametersByName.statement || request.parametersByName.stmt
+      return statement.value
+    }
+  }
+}
+
+function addConnectionTags (span, connectionConfig) {
+  span.setTag('out.host', connectionConfig.server)
+  span.setTag('out.port', connectionConfig.options.port)
+}
+
+function addDatabaseTags (span, connectionConfig) {
+  span.setTag('db.user', connectionConfig.userName || connectionConfig.authentication.options.userName)
+  span.setTag('db.name', connectionConfig.options.database)
+  span.setTag('db.instance', connectionConfig.options.instanceName)
+}
+
+function addProcIdTags (span, request) {
+  if (!request.sqlTextOrProcedure.match(procnameRegex)) return
```

Comment:
`procnameRegex.test(request.sqlTextOrProcedure)` would be more robust as it works with `undefined` and returns a boolean instead of a match which is not needed when you don't need capturing groups.